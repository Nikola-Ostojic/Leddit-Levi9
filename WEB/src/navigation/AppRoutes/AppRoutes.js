import React from "react";
import { Switch, Route } from "react-router-dom";

import HomePage from "pages/HomePage";
import MoviesPage from "pages/MoviesPage";
import CreateThreadPage from "pages/CreateThreadPage";
import MovieDetailsPage from "pages/MovieDetailsPage";
import ThreadDetailsPage from "pages/ThreadDetailsPage";
import NotFoundPage from "pages/NotFoundPage";
import AuthPage from "pages/AuthPage";
import "./AppRoutes.css";

import PrivateRoute from "../AppRoutes/PrivateRoute";

const AppRoutes = () => (
  <div className="content">
    <Switch>
      <Route exact path="/" component={HomePage} />
      <Route path="/threadDetails/:id?" component={ThreadDetailsPage} />
      <PrivateRoute path="/movies" component={MoviesPage} />
      <PrivateRoute path="/movieDetails/:id?" component={MovieDetailsPage} />
      <PrivateRoute path="/createThread" component={CreateThreadPage} />
      <Route path="/auth" component={AuthPage} />
      <Route path="*" component={NotFoundPage} />
    </Switch>
  </div>
);

export default AppRoutes;
