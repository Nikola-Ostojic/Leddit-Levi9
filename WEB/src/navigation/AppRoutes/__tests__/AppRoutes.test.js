import React from "react";
import { shallow } from "enzyme";

import AppRoutes from "../AppRoutes";

describe("AppRoutes test", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<AppRoutes />);
    expect(wrapper).toMatchSnapshot();
  });
});
