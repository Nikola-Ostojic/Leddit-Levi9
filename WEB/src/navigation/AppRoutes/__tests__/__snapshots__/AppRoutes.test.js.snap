// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoutes test Should match snapshot 1`] = `
<div
  className="content"
>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(withRouter(HomePage))",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(withRouter(ThreadDetailsPage))",
          "type": [Function],
        }
      }
      path="/threadDetails/:id?"
    />
    <Connect(PrivateRoute)
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(withRouter(MoviesPage))",
          "type": [Function],
        }
      }
      path="/movies"
    />
    <Connect(PrivateRoute)
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MovieDetailsPage)",
          "type": [Function],
        }
      }
      path="/movieDetails/:id?"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/createThread"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": Object {
            "$$typeof": Symbol(react.memo),
            "compare": [Function],
            "type": [Function],
          },
          "compare": null,
          "displayName": "Connect(Component)",
          "type": [Function],
        }
      }
      path="/auth"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</div>
`;
